File: platform/kui/cli/Help.kt
package kui.cli

/**
 * Standardized help message formatter for KUI CLI.
 */
object Help {
    val TEXT: String = buildString {
        appendLine(KuiVersion.DISPLAY_NAME + " - Pure Kotlin Application Platform Toolchain")
        appendLine()
        appendLine("Usage:")
        appendLine("  kui [command] [options]")
        appendLine()
        appendLine("Commands:")
        appendLine("  new <name>     create project")
        appendLine("  doctor         verify environment")
        appendLine("  info           show project information")
        appendLine("  build          compile/package")
        appendLine("  run            build + install + launch")
        appendLine("  install        install current APK")
        appendLine("  launch         launch installed app")
        appendLine("  test           run tests")
        appendLine("  clean          remove generated build output")
        appendLine("  bench          benchmarks")
        appendLine("  profile        profiling report")
        appendLine("  ui-tree        print runtime UI tree later")
        appendLine("  help           show help for commands")
        appendLine()
        appendLine("Options:")
        appendLine("  -v, --version  print version information")
        append("  -h, --help     print this help message")
    }

    fun printHelp() {
        println(TEXT)
    }
}

File: crates/kui-cli/src/cli/help.rs
use super::command::Command;
use super::version::DISPLAY_NAME;

pub struct Help;

impl Help {
    pub fn general_help() -> String {
        let mut out = String::new();
        out.push_str(&format!("{DISPLAY_NAME} - Pure Kotlin Application Platform Toolchain\n\n"));
        out.push_str("Usage:\n  kui [command] [options]\n\nCommands:\n");

        for cmd in Command::all() {
            let display_name = if *cmd == Command::New {
                "new <name>".to_string()
            } else {
                cmd.command_name().to_string()
            };
            out.push_str(&format!("  {:<14} {}\n", display_name, cmd.description()));
        }

        out.push_str("\nOptions:\n");
        out.push_str("  -v, --version  print version information\n");
        out.push_str("  -h, --help     print this help message\n");
        out
    }

    pub fn command_help(cmd: Command) -> String {
        format!("{} - {}\n\nUsage:\n  {}\n", cmd.command_name(), cmd.description(), cmd.usage())
    }

    pub fn print_help(target: Option<Command>) {
        if let Some(cmd) = target {
            print!("{}", Self::command_help(cmd));
        } else {
            print!("{}", Self::general_help());
        }
    }
}
